src/key.rs

```diff
@@ -915,6 +916,313 @@ impl Display for Key {
             __Nonexhaustive => unreachable!(),
         }
     }
-} 
+}
+
+impl FromStr for Key {
+    type Err = UnrecognizedKeyError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        use Key::*;
+        match s {
+            ch if ch.chars().take(2).count() == 1 => Ok(Character(ch.to_string())),
```

Comment:
A key string can also consist of a base character and multiple combining characters.
https://w3c.github.io/uievents-key/#keys-unicode
Not sure if it is worth supporting.

Additionally Control Characters should be excluded.

Reply:
I could add the exact check if string is a _key string_, but I'm not sure how do you fell with adding new dependency (I would use https://docs.rs/unic-ucd-category/0.9.0/unic_ucd_category/enum.GeneralCategory.html )

Reply:
Yeah, the dependency would be quite big for such a small feature.

How about checking that:

1. there are no control characters in the string (`char::is_control`)
2. No characters besides the first are in the ASCII range (`char::is_ascii`)

This prevents common mistakes like mistyped keynames (e.g. `Ennter`) from being recognized as characters while allowing all characters specified. If the need arises we can add the dependency later.